packages/core/src/storage.ts

```diff
@@ -71,43 +58,46 @@ export const localStorageGetter = (
       }
     }
   }
-  if (clientStorageType === 'custom') {
-    if (clientStorage.getItem && clientStorage.removeItem) {
-      return clientStorage.getItem
+
+  if (clientStorage) {
```

Comment:
I think we should perform an early exit here if `clientStorage` is undefined. We can make the code more readable if we reduce the depth of blocks IMO.